pscheduler-core/pscheduler-core/metrics

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/python3
+#
+# Output pScheduler statistics in the Prometheus format
+# To be used with node-exporter textfile collector
+#
+
+import optparse
+import pscheduler
+import json
+
+pscheduler.set_graceful_exit()
+
+# Known stats URL
+# Use an empty list for state when there are no subsequent stats
+
+pscheduler_stats = {
+    "paused": {
+        "url": "stat/control/pause",
+        "prometheus-name": "perfsonar_pscheduler_paused",
+        "desc": "pScheduler is paused",
+        "type": "gauge",
+        "state": [ "" ]
+        },
+    "archiving": {
+        "url": "stat/archiving/",
+        "prometheus-name": "perfsonar_pscheduler_archiving",
+        "desc": "pScheduler archiving queue",
+        "type": "gauge",
+        "state": [ "backlog", "upcoming" ]
+        },
+    "http-queue": {
+        "url": "stat/http-queue/",
+        "prometheus-name": "perfsonar_pscheduler_http_queue",
+        "desc": "pScheduler HTTP queue",
+        "type": "gauge",
+        "state": [ "backlog", "length" ]
+        },
+    "runs": {
+        "url": "stat/runs/",
+        "prometheus-name": "perfsonar_pscheduler_runs",
+        "desc": "Number of pScheduler runs in the different states",
+        "type": "gauge",
+        "state": [ "pending", "on-deck", "running", "cleanup", "finished", "overdue", "missed", "failed", "preempted", "nonstart" ]
+        }
+    }
+
+#
+# Gargle the arguments
+#
+
+class VerbatimParser(optparse.OptionParser):
+    def format_epilog(self, formatter):
+        return self.epilog
+
+opt_parser = VerbatimParser(
+    usage="Usage: %prog [options]",
+    epilog=
+"""
+Examples:
+
+  metrics
+      Output pScheduler statistics as a JSON object.
+
+  metrics --format prometheus
+      Output pScheduler statistics in a format suitable for
+      Prometheus node-exporter textfile collector
+"""
+    )
+opt_parser.disable_interspersed_args()
+
+opt_parser.add_option("--format",
+                      help="The output format to use, either json (default) or prometheus",
+                      default="json",
+                      action="store", type="string",
+                      dest="format")
+opt_parser.add_option("--timeout", "-W",
+                      help="How long to wait for the server to respond, in seconds (default 5)",
+                      default=5,
+                      action="store", type="int",
+                      dest="timeout")
+
```

Comment:
I recommend adding a `--host` option (see other programs, like `plot-schedule`) and using it in forming URLs for the API.  See comment below.